src/tiledb/cloud/vector_search/file_ingestion.py

```diff
@@ -0,0 +1,326 @@
+from typing import Dict, List, Optional, Sequence
+
+from tiledb.cloud import dag
+from tiledb.cloud.utilities import as_batch
+
+
+def ingest_files_dag(
+    file_dir_uri: str,
+    index_uri: str,
+    file_name: Optional[str] = None,
+    acn: Optional[str] = None,
+    config=None,
+    namespace: Optional[str] = None,
+    verbose: bool = False,
+    trace_id: Optional[str] = None,
+    # Index creation params
+    create_index: bool = True,
+    index_type: str = "IVF_FLAT",
+    index_creation_kwargs: Dict = {},
+    # DirectoryTextReader params
+    glob: str = "**/[!.]*",
+    exclude: Sequence[str] = (),
+    suffixes: Optional[Sequence[str]] = None,
+    text_splitter: str = "RecursiveCharacterTextSplitter",
+    text_splitter_kwargs: Optional[Dict] = {
+        "chunk_size": 500,
+        "chunk_overlap": 50,
+    },
+    # SentenceTransformersEmbedding params
+    model_name_or_path: str = "BAAI/bge-small-en-v1.5",
+    # Index update params
+    index_timestamp: int = None,
+    workers: int = -1,
+    worker_resources: Dict = None,
+    worker_image: str = None,
+    extra_worker_modules: Optional[List[str]] = None,
+    driver_resources: Dict = {"cpu": "2", "memory": "8Gi"},
+    driver_image: str = None,
+    extra_driver_modules: Optional[List[str]] = None,
+    max_tasks_per_stage: int = -1,
+    embeddings_generation_mode: dag.Mode = dag.Mode.LOCAL,
+    embeddings_generation_driver_mode: dag.Mode = dag.Mode.LOCAL,
+    vector_indexing_mode: dag.Mode = dag.Mode.LOCAL,
+    index_update_kwargs: Dict = {"files_per_partition": 100},
+):
+    """
+    Ingest files into a vector search text index.
+
+    :param file_dir_uri: directory of the files to be loaded. For individual files,
+        also pass the `file_name` param.
+    :param index_uri: URI of the vector index to load files to.
+    :param file_name: Name of the file to be loaded.
+    :param acn: Access Credentials Name (ACN) registered in TileDB Cloud (ARN type),
+        defaults to None.
+    :param config: config dictionary, defaults to None.
+    :param namespace: TileDB-Cloud namespace, defaults to None.
+    :param verbose: verbose logging, defaults to False.
+    :param trace_id: trace ID for logging, defaults to None.
+    # Index creation params
+    :param create_index: If true, creates a new vector search index.
+    :param index_type: Vector search index type ("FLAT", "IVF_FLAT").
+    :param index_creation_kwargs: Arguments to be passed to the index creation method
+    # DirectoryTextReader params.
+    :param glob: Glob pattern relative to the specified path by default set to pick up
```

Comment:
Lets standardize the parameters, we use `include` and `exclude` in vcf, imaging and geospatial ingestion functions.

Reply:
This will require some changes in the vector search library. Is this ok if I address it in a separate PR?

Reply:
Consolidated the arguments. This is blocking on the release of https://github.com/TileDB-Inc/TileDB-Vector-Search/pull/283